pkg/proc/target_group.go

```diff
@@ -0,0 +1,125 @@
+package proc
+
+import (
+	"fmt"
+	"strings"
+)
+
+// TargetGroup reperesents a group of target processes being debugged that
+// will be resumed and stopped simultaneously.
+// New targets are automatically added to the group if exec catching is
+// enabled and the backend supports it, otherwise the group will always
+// contain a single target process.
+type TargetGroup struct {
+	targets []*Target
+	procs   []ProcessInternal
```

Comment:
The `Target` type will already have it's associated `ProcessInternal` why store them separately?

Reply:
Trying to not reallocate it every time, but it's an optimization and I didn't really measure it so I'm changing it.